### ISSUE TITLE
`cargo docs` nightly fails.

### ISSUE THREAD

**Issue by HaoYang670** [apache/arrow-rs#2385]
**Describe the bug**

**To Reproduce**
```
cargo doc --document-private-items --no-deps --workspace --all-features
```
using the nightly tool chain.

Get error
```
warning: unnecessary `unsafe` block
   --> arrow/src/compute/kernels/arithmetic.rs:369:17
    |
369 |                 unsafe { result.typed_data_mut().chunks_exact_mut(64) };
    |                 ^^^^^^ unnecessary `unsafe` block
    |
    = note: `#[warn(unused_unsafe)]` on by default

warning: unnecessary `unsafe` block
   --> arrow/src/compute/kernels/arithmetic.rs:417:17
    |
417 |                 unsafe { result.typed_data_mut().chunks_exact_mut(lanes) };
    |                 ^^^^^^ unnecessary `unsafe` block

warning: unnecessary `unsafe` block
    --> arrow/src/compute/kernels/comparison.rs:1700:25
     |
1700 |     let result_chunks = unsafe { result.typed_data_mut() };
     |                         ^^^^^^ unnecessary `unsafe` block
...
```

**Expected behavior**

**Additional context**

**Comment by HaoYang670** [comment]
I am not sure whether this is related to the new released `rust 1.65.0 nightly`

**Comment by HaoYang670** [comment]
Close this issue as I don't find the error is reproduced in other PR's CI.
